### PR TITLE
Add regen-cabal.sh

### PR DESCRIPTION
**Pull request by hasufell**
Since there seems to be still confusion on how to regenerate cabal project files, maybe this makes it easier.

* [x] Ran `ShellCheck`
* [x] tested on linux
* [x] tested on mac

### CODE REVIEW

File: scripts/regen-cabal.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+SOURCE="$0"
+SCRIPT_DIR="$(dirname "$(realpath "${SOURCE}")")"
+
+cd "${SCRIPT_DIR}"/..
+
+mkdir -p .tmp/bin
+export PATH="$(realpath .tmp/bin):$PATH"
+
+vercomp() {
```
Comment (hasufell) [inline scripts/regen-cabal.sh:66]: This is a copy-paste... bash doesn't have libraries :)

File: scripts/regen-cabal.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+stack2cabal_ver="1.0.11"
+
+SOURCE="$0"
+SCRIPT_DIR="$(dirname "$(realpath "${SOURCE}")")"
+
+cd "${SCRIPT_DIR}"/..
+
+mkdir -p .tmp/bin
+export PATH="$(realpath .tmp/bin):$PATH"
```
Comment (hasufell) [inline scripts/regen-cabal.sh]: So we currently always prefer a local copy of the binary, if there is one (over a system one). I think that's fair.

**Review by rvl — COMMENTED** [review]
The reason we have Nix is so that you don't need to write bash scripts like this.

**Comment by hasufell** [comment]
> The reason we have Nix is so that you don't need to write bash scripts like this.

I've had more problems with nix than with bash scripts. And I don't want regenerating cabal files take up 6 GB of memory, require downloading half an OS and then deal with all sorts of esoteric errors.

But feel free to rewrite this in nix, because I'm using `stack2cabal` directly anyway. Otherwise I want to move forward with this, so ppl don't ask me every time on how to regenerate those files.

**Comment by KtorZ** [comment]
Closing this one. We do not want to have to maintain yet-another-script, especially bash. What would be good would be to extend a bit the instructions [in the wiki](https://github.com/input-output-hk/cardano-wallet/wiki/Building#syncing-stack-and-cabal-dependencies) regarding the re-generation of cabal.project.

**Pull request closed without merging**